src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -196,6 +208,46 @@ public void ConnectHeaders_SupplyHeadersThenSupplyMore_Both_Tip_PeerId_Maps_Shou
             Assert.True(peerTipsByPeerIdBefore.Values.Count == peerTipsByPeerIdAfter.Values.Count);
         }
 
+        /// <summary>
+        /// Issue 5 @ Create chained header tree component #1321
+        /// 2 peers should supply headers - one of the peers creates a fork.
+        /// Make sure that ChainedBlock is not created more than once for the same header.
+        /// Check that next pointers were created correctly.
+        /// </summary>
+        [Fact]
+        public void ConnectHeaders_HeadersFromTwoPeersWithFork_ShouldCreateBlocksForNewHeaders()
+        {
+            // Setup
+            var ctx = new TestContext();
+            ChainedHeaderTree cht = ctx.CreateChainedHeaderTree();
+            ChainedHeader chainTip = ctx.ExtendAChain(10);
+            cht.Initialize(chainTip, true);
+            ctx.ChainStateMock.Setup(s => s.ConsensusTip).Returns(chainTip);
+
+            // Default peer is on tip
+            Assert.True(cht.GetPeerTipsByPeerId()[ChainedHeaderTree.LocalPeerId].Equals(chainTip.HashBlock));
+
+            // Peer 1 forks two blocks back
+            BlockHeader forkedBlock = ctx.GetNewBlock(chainTip.Previous.Previous).Header;
+
+            cht.ConnectNewHeaders(1, new List<BlockHeader>() { forkedBlock });
+
+            uint256 peerOneTipHash = cht.GetPeerTipsByPeerId()[1];
+            uint256 hashOfForkedBlock = forkedBlock.GetHash();
+
+            ChainedHeader chainedHeaderOfForkedBlock = cht.GetChainedHeadersByHash()[hashOfForkedBlock];
+            ChainedHeader fork = chainTip.FindFork(chainedHeaderOfForkedBlock);
+
+            fork.Should().NotBeNull();
```

Comment:
Please use normal assert

Reply:
I don't agree with that.  Fluent asserts have a very nice syntax that flows very nicely.

Assert.Equal is error prone and makes you think too much.    Fork should not be null. couldn't be any clearer could it ?

Reply:
Fluent stuff hides the logic behind the comment so people read the intent, not what the thing does. 
Imho it's just adds another level ob abstraction and before starting to read the test I first will need to read all the methods with nice names that encapsulate it. 

This line from the next comment is a good example:

```
peerOneTipHash.Should().Be(hashOfForkedBlock, "Peer should be on the fork but wasn't");
```

It reads good but what `Should` and `Be` methods do? So I first will go and press F12 on should, read it, go back, go to Be, read it, go back, read the line again. 
Assert.Equal(val1, val2) is much easier to read.

Reply:
No, you won't - you just read and accept :)  `Peer one tip hash should be hash of forked block`  -fails if it isn't

There is no requirement to go to the implementation of that - but feel free to check it once. It is a well known nuget package and used by millions of projects.

Reply:
you find `Assert equal expected actual`  is easier to read than `actual should be expected` ?

Reply:
I find `Assert equal expected actual` and  `actual should be expected` equal easy to read.

But I find 

`Assert.Equal(var1,var2)` easier to read than `var1.Should().Be(var2)`

Reply:
```
But I find

Assert.Equal(var1,var2) easier to read than var1.Should().Be(var2)
```
No you don't - because you got var1 and var 2 the wrong way round in Assert.Equal - which is the exact reason why a fluent syntax is more logical and less error prone.

Reply:
I suggest that you might jus be conditioned to reading Assert a lot, which is fine, but don't rule out fluent assertions as a bad thing just because you have seen Assert.Equal a lot until now.

Reply:
I base those preferences on several tries to read current integration tests in BDD style. Way harder for me to see the whole picture that way. 

So it's about personal preference- some find reading black letters on white paper better, some like white letters on black background.

Reply:
For me personally `var1.Should().Be(var2)` is quite unreadable. It is incorrectly combining English language with C#. It reads wrong because it suggests reference to quality instead of value equality. It is absolutely illogical from my point of view.

Reply:
I prefer `Should` instead of `FluentAssertions`  but fluent assertions was already used in the solution in the DNS project so suggesting to bring in a different `should` extensions library would have been rejected by the code style police. Hence a happy medium of using something that is already there.

Reply:
ha - this isn't even my code :)

Reply:
Just saying that I really like FluentAssertions over Assert, Assert is too symetrical, please don't deprive us of that ! :)

Reply:
>Just saying that I really like FluentAssertions over Assert, Assert is too symetrical, please don't deprive us of that ! :)

No one does. BDD is allowed for the integration tests. For the rest of the code- normal codestyle is required

Reply:
BDD is not a code style. To many developers, bdd is normal, not two things that are mutually exclusive. BDD is behaviour driven development - ie driving out the design of the code through specifying behaviours.  You would never say.... we don’t do tdd style tests, we do normal ones so you should never say we don’t do BDD style tests we do normal.   If there is a policy in place that says you aren’t allowed to write expressively worded tests with methods that describe the exact behaviour under test, without using fluent assertions, please show us that policy.

And in reference to the above, Should is unrelated to whether you use TDD, BDD or Test-After.